clippy_lints/src/strings.rs

```diff
@@ -173,16 +176,63 @@ fn is_add(cx: &LateContext<'_>, src: &Expr<'_>, target: &Expr<'_>) -> bool {
     }
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Check if the string is transformed to byte array and casted back to string.
+    ///
+    /// **Why is this bad?** It's unnecessary, the string can be used directly.
+    ///
+    /// **Known problems:** None
+    ///
+    /// **Example:**
+    /// ```rust
+    /// let string = "Hello World!";
+    /// let string2 = std::str::from_utf8(&string.as_bytes()[6..11]).unwrap();
```

Comment:
We can inline `string` to shorten the example. Also `string2` doesn't need to be named, we can just use `_`.

Reply:
Done